diff --git a/index.js b/index.js
index 23384ea16bb848616886f356d8d69f94325535bf..41a7bf6ccc7f1d6f47197536c03f14920f41ae55 100644
--- a/index.js
+++ b/index.js
@@ -127,5 +127,11 @@ class Extractor {
     if (symlink) {
       const link = await getStream(readStream)
+      const canonicalLink = path.resolve(path.dirname(dest), link)
+      const relativeLink = path.relative(this.opts.dir, canonicalLink)
+      if (path.isAbsolute(link) || relativeLink === '..' || relativeLink.startsWith(`..${path.sep}`)) {
+        throw new Error(`Out of bound symlink target "${link}" found while processing file ${entry.fileName}`)
+      }
+
       debug('creating symlink', link, dest)
       await fs.symlink(link, dest)
     } else {
